File: JavaScript/4/4_3.js
const userSession = {
  isLoggedIn: true,
  token: "abc123"
};
//Fucntion will log out the user by updating the session state.
function logout() {
  userSession.isLoggedIn = false;
  userSession.token = null;
  console.log("User logged out.");
}

logout();

File: C#/4/4_3.cs
class UserSession
{
    public bool IsLoggedIn { get; set; } = true;
    public string Token { get; set; } = "abc123";
}

UserSession userSession = new UserSession();

UserSession Logout(UserSession currentSession)
{
    var newSession = new UserSession
    {
        IsLoggedIn = false,
        Token = null
    };
    Console.WriteLine("User logged out.");
    return newSession;
}

userSession = Logout(userSession);
